script.io.benchmark/default.py

```diff
@@ -0,0 +1,154 @@
+ #   Copyright (C) 2017 Lunatixz
+#
+#
+# This file is part of I/O Benchmark.
+#
+# I/O Benchmark is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# I/O Benchmark is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with I/O Benchmark.  If not, see <http://www.gnu.org/licenses/>.
+
+import os, sys, monkeytest
+import xbmc, xbmcgui, xbmcplugin, xbmcvfs, xbmcaddon
+
+
+# Plugin Info
+ADDON_ID       = 'script.io.benchmark'
+REAL_SETTINGS  = xbmcaddon.Addon(id=ADDON_ID)
+ADDON_NAME     = REAL_SETTINGS.getAddonInfo('name')
+ADDON_PATH     = (REAL_SETTINGS.getAddonInfo('path').decode('utf-8'))
+ADDON_VERSION  = REAL_SETTINGS.getAddonInfo('version')
+SETTINGS_LOC   = REAL_SETTINGS.getAddonInfo('profile')
+ICON           = os.path.join(ADDON_PATH, 'icon.png')
```

Comment:
The end-result (the path you're getting) yes, the way the path is built/retrieved...
...no idea actually.

Either way. Kodi (which provides the whole framework for us to work with) *should* know best how and where to retrieve things like icons etc. hence why should we do it?

In case core changes we (python add-on devs) can be like *"none of our business"* which is why we leverage things like these to whom has to deal with it anyways :)